src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,157 @@
+package duke;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+
+import duke.exception.DukeTaskNotFoundException;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Todo;
+
+
+/**
+ * TaskList class.
+ * Handle tasks in a list.
+ */
+public class TaskList {
+
+    protected ArrayList<Task> arrayList;
+
+    /**
+     * Constructor for TaskList.
+     * @param arrayList ArrayList to be wrapped around TaskList.
+     */
+    public TaskList(ArrayList<Task> arrayList) {
+        this.arrayList = arrayList;
+    }
+
+    /**
+     * Marks the task and returns it.
+     * @param index Index of task in the list.
+     * @return Marked task.
+     */
+    public Task mark(Integer index) {
+        arrayList.get(index - 1).markAsDone();
+        return arrayList.get(index - 1);
+    }
+
+    /**
+     * Unmarks the task and returns it.
+     * @param index Index of task in the list.
+     * @return Unmarked task.
+     */
+    public Task unmark(Integer index) {
+        arrayList.get(index - 1).markAsUndone();
+        return arrayList.get(index - 1);
+    }
+
+    /**
+     * Deletes task from the list and returns it.
+     * @param index Index of task in the list.
+     * @return Deleted Task.
+     */
+    public Task delete(Integer index) {
+        Task t = arrayList.get(index - 1);
```

Comment:
Would the variable name t be ambiguous? Perhaps a more intuitive name such as "task" could be used. I noticed the same issue in several other places too.